staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -81,6 +81,12 @@ type ListMeta struct {
 	// identical to the value in the first response, unless you have received this token from an error
 	// message.
 	Continue string `json:"continue,omitempty" protobuf:"bytes,3,opt,name=continue"`
+
+	// the number of total matching objects in the storage starting from
+	// the continue key in the listOptions (not the continue in this list
+	// metadata). If the continue key in the listOptions is empty, count is
```

Comment:
If no pagination was requested, then there's no need for the server to set this, is there?

Consider calling this field "RemainingItems". The description can then say "If this list is partial (due to pagination), then this field counts how many more items there are in the collection."

We can do math on the count etcd gives us to make sure the items sent in the list are not included in the remainingItems field. Mostly I just think it's difficult to understand at the moment.

Reply:
Given either a total count or a remaining count value, a client can calculate the other. So probably best to only introduce one of them. Remaining could be implemented like @lavalamp suggests, a total count could be implemented by getting the count from etcd on the initial list call (with or without a limit) and then remembering somehow (e.g. hiding it in the continue token) since subsequent etcd list calls will start at a range offset and won't provide the same total. Having a total count seems simple and consistent to me, although it's not actually needed when no limit is provided since it can be calculated.

Reply:
I think remaining is more useful, and more convenient, since we don't have to make additional etcd calls.

Since it's possible to start in the middle of a collection, the total count isn't very helpful: you can't take a total count and subtract the number of items you've seen/processed to get the remaining number of items, because items could have been added or removed prior to the page you're getting.

So, the useful thing for clients is to track how many items they have processed, and let the server tell them how many more items there are at the moment. You can use this to e.g. estimate a completion time.

Reply:
Continue provide a point-in-time view of a list, any items added or removed once the paginated list is initiated are not visible, so the item count stay the same across the entire paginated listing operation. No additional etcd calls are needed. But remaining will certainly work, count just feels like a more fundamental base fact to me.

Reply:
No-- it's not clear in the documentation above, but if your continue key has expired, the system will give you a new one along with the error message. You can use this feature to resume in the middle of a long list.